### ISSUE TITLE
Chapter-by-Chapter Scrolling for Bible

### ISSUE THREAD

**Issue by Breadoggo** [AndBible/and-bible#3298]
I'm always frustrated when I scroll through a book of the Bible because, although the app shows where a chapter ends, it is not very clear. This makes it difficult to recognize the transition between chapters and can be confusing during reading.

  I would like the app to have an option that makes it so that when I scroll through the Bible, I can only scroll to the end of each chapter. After reaching the end of a chapter, I should have to manually go to the next chapter(using the already existing feature of swiping to the right or left) to continue reading. This would ensure clear recognition of the end of one chapter and the start of another.

  Since this feature would be optional, it wouldn't inconvenience any users, as they could choose their preferred scrolling method.

  I've provided some images to illustrate how this feature would work: 
![IMG_20240731_100911](https://github.com/user-attachments/assets/59a09163-ab00-4396-85dd-613e64ae9fa2)
![IMG_20240731_100954](https://github.com/user-attachments/assets/d3e19824-ef39-4133-923e-ab2965947cdf)
![IMG_20240731_101010](https://github.com/user-attachments/assets/83e35577-5f96-4f36-828d-c1928243566b)

**Comment by Noitarud** [comment]
Some bible modules have titles for the stories in bold, so why not try another translation? The NASB has headings.

I would like to point out there were no "chapters" when that was written, chapters are even obstructive, try reading Mat25:1 without the interruption.

Psalms is about the only exception, and most of those have headings (AB shows as bold in KJV), the ones that do not probably run from the previous, an example is Psalm103-107, which I think even traverses books (they were kept in five volumes, most volumes have closing phrase(s)) - but maybe the psalms being longer idea is just my opinion.

**Comment by tuomas2** [comment]
Implementation plan:
- create an option to disable automatic infinite scrolling. 
- If disabled, show button on the bottom of screen (for bibles) which need to be tapped to see more content.

Estimate: 2h.

**Comment by meichthys** [comment]
I've thought about this as well. I've noticed that when you flick scroll backwards in a Bible, the scrolling stops at the previous chapter, then you can flick again and the scrolling will continue until the next chapter. This is kind of nice, however, this option is not available when scrolling forward in the Bible. instead, a flick scroll continues until the memory buffer runs out (or indefinitely).

https://github.com/user-attachments/assets/8948548e-7a92-45f3-bb80-cc2961c69544

I wonder if this flick scrolling until the next chapter option was available for forward scrolling if that would be another potential resolution to this issue.

@Breadoggo would this improve the situation, or would you still prefer to only be able to scroll within a single chapter?

**Comment by tuomas2** [comment]
Flick when going backwards is technical "issue": it does not automatically load content backwards and only does so if user scrolls backwards to a certain point. I remember it was not that easy to make it load content to the front of the page without causing such a flick. 

Of course such a flick could be artificially created when scrolling forward too, but I'm not sure/convinced that it would be pleasant. But optionally, maybe...

**Comment by meichthys** [comment]
Yes, if implemented it should definitely be an option, not a default.

**Comment by meichthys** [comment]
@tuomas2 
Should the forward flick scrolling option be a separate feature request, or can this issue be resolved by adding that artificial stop when scrolling forward in the text? It seems like the artificial gorward-glick pause would improve the OP's situation over the default implementation.

I'd like to support this forward flick feature if possible.

**Comment by tuomas2** [comment]
This is something that would need a bit of experimentation to figure out what feels most natural. The option could go in Text Display Settings or App Preferences.

I've actually already experimented with related approaches while working on the LLM feature — some of those experiments involved optionally disabling infinite scrolling. So there's some groundwork there to build on.
